drivers/i2s/i2s_stm32_sai.c

```diff
@@ -304,15 +310,16 @@ static int i2s_stm32_sai_dma_init(const struct device *dev)
 	hdma->Init.TransferEventMode = DMA_TCEM_BLOCK_TRANSFER;
 #endif
 
-#if defined(CONFIG_SOC_SERIES_STM32H7X)
+#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32F4X)
 	hdma->Init.FIFOMode = DMA_FIFOMODE_DISABLE;
 #endif
 
 	if (stream->dma_cfg.channel_direction == (enum dma_channel_direction)MEMORY_TO_PERIPHERAL) {
 		hdma->Init.Direction = DMA_MEMORY_TO_PERIPH;
 
 #if !defined(CONFIG_SOC_SERIES_STM32H7X) && !defined(CONFIG_SOC_SERIES_STM32L4X) &&                \
-	!defined(CONFIG_SOC_SERIES_STM32G4X) && !defined(CONFIG_SOC_SERIES_STM32L5X)
+	!defined(CONFIG_SOC_SERIES_STM32G4X) && !defined(CONFIG_SOC_SERIES_STM32L5X) &&            \
+	!defined(CONFIG_SOC_SERIES_STM32F4X)
```

Comment:
Could we replace all these defined with a single `#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32u5_dma)` ?
That would probably make adding other series easier.